torch/distributed/_tensor/sharding_prop.py

```diff
@@ -124,10 +125,17 @@ def _wrap_output_spec_tensor_meta(
                         spec.tensor_meta = output_tensor_meta_i
 
     def propagate(self, op_info: OpInfo) -> None:
-        output_sharding = self.propagate_op_sharding(op_info.schema)
+        # We cannot use an lru cache if we know that inputs will have dynamic shapes,
+        # because SymInts are not hashable.
+        # This is generally ok because this only happens during tracing in torch.compile,
+        # and tracing does not need to be as fast as eagermode DTensor usages.
+        if _are_we_tracing():
```

Comment:
welp, this isn't enough. It's possible to have the current `op_info.schema` hold onto SymInts, even when `_are_we_tracing()` returns false (because neither a FakeTensorMode or TorchProxyDispatchMode are active).

I'm going to update this check to instead just directly poke at the `op_info.schema` and check if it has any SymInts on it.